core/translate.rs

```diff
@@ -1403,7 +1406,7 @@ fn update_pragma(name: &str, value: i64, header: Rc<RefCell<DatabaseHeader>>, pa
 }
 
 fn maybe_apply_affinity(col: &Column, target_register: usize, program: &mut ProgramBuilder) {
-    if col.ty == crate::schema::Type::Real {
+    if col.ty == crate::schema::Type::Real && !program.is_translating_cond {
```

Comment:
This looks like a hack to be honest. I think the problem here is that we're talking the type from a _column_, but actually we want a different type depending on where in an expression we use it. So maybe have `maybe_apply_affinity` accept a `type` parameter and make sure we pass the right type depending on the expression context.

Reply:
yeah @penberg agreed this PR is more of a hack, this might not be a good idea. Might wait until we refactor `translate.rs` before we look at it again.

Reply:
Also noticed that the `select: &Select` parameter drilling in `translate_expr` or similar functions, only to be used when resolving table name. Would it be a good idea to store the current `Select` in `ProgramBuilder`?

Reply:
Yeah, although I am sucker for merging early, I would actually prefer not to do this. It's introducing dependency to program builder, which can be hard to get rid of later.

Reply:
ah ok, maybe I'll turn this into an issue to track, close this MR first and revisit in the future